website/files/models/osfstorage.py

```diff
@@ -238,14 +239,20 @@ def add_tag(self, tag, auth, save=True, log=True):
     def remove_tag(self, tag, auth, save=True, log=True):
         from website.models import Tag, NodeLog  # Prevent import error
         tag = Tag.load(tag)
-        if tag and tag in self.tags and not self.node.is_registration:
+        if not tag:
+            raise InvalidTagError
+        elif tag not in self.tags:
+            raise TagNotFoundError
+        elif self.node.is_registration:
```

Comment:
That's a good point, and this change would make the behavior of the project and file endpoints slightly more consistent.

Commit pushed.